### PR TITLE
Fix issues with WELOPEN and COMPDAT entries at the same date

### PR DESCRIPTION
**Pull request by wouterjdb**
`WELOPEN` previously would have overwritten all actions in at the current date; now, the order of the keywords inside of the date-block dictates what happens as it would in a simulation run.

The following SCHEDULE section,
```
DATES
 x x x /

WELOPEN
 OP1 SHUT /
/
COMPDAT
 OP1 1 1 1 OPEN /
/
```
now produces one `COMPDAT` line with `OPEN` instead of `SHUT` one.

### CODE REVIEW

File: ecl2df/compdat.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ def deck2dfs(deck, start_date=None, unroll=True):
                     rec, "COMPDAT", renamer=COMPDAT_RENAMER
                 )
                 rec_data["DATE"] = date
+                rec_data["KEYWORD_ID"] = idx
```
Comment (berland) [inline ecl2df/compdat.py:96]: `KEYWORD_ID` -> `KEYWORD_IDX` ?

File: ecl2df/compdat.py [hunk 2]
```diff
@@ -155,6 +157,9 @@ def deck2dfs(deck, start_date=None, unroll=True):
     if unroll and not welsegs_df.empty:
         welsegs_df = unrolldf(welsegs_df, "SEGMENT1", "SEGMENT2")
 
+    if "KEYWORD_ID" in compdat_df.columns:
```
Comment (berland) [inline ecl2df/compdat.py:160]: Should we drop it, or could it become handy to keep for some other use later?

Comment (berland) [reply ecl2df/compdat.py:160]: (who knows what problems will creep  up when we finally manage to auto-merge these three dataframes)

Comment (berland) [reply ecl2df/compdat.py:160]: thinking of issue #6

Comment (wouterjdb) [reply ecl2df/compdat.py]: It will definitely come in handy - for the same reasons. But I'd say we can mention that in that issue (extend usage of `KEYWORD_IDX`). If I would not return `KEYWORD_IDX` I'd most likely break many other things (at least some tests).

**Comment by berland** [comment]
Can we have a test where WELOPEN only changes a portion of a well?

File: ecl2df/compdat.py (hunk 2 above)
Comment (berland) [reply ecl2df/compdat.py]: ok!

**Comment by berland** [comment]
Can we have a test where WELOPEN is repeated, first SHUT and then OPEN (or reverse) in sequence, with no DATES or COMPDAT in between them? Probably valid E100, although it is strange  setup.

**Comment by wouterjdb** [comment]
Added the tests 🧪

**Comment by berland** [comment]
It looks good. My hunch is that there are more corner cases to cater for, for which tests are the only gold. Can you look a little bit for more sharp corners?

File: tests/test_welopen.py [hunk 3]
```diff
@@ -331,6 +331,60 @@
...
+     1 MAY 2001 /
+    /
+
+    COMPDAT
+     'OP1' 1 1 1 1 'OPEN'  /
+    /
+
+    WELOPEN
+     'OP1' 'OPEN' /
+     'OP1' 'SHUT' /
```
Comment (berland) [inline tests/test_welopen.py:346]: nice!

**Comment by wouterjdb** [comment]
Added two more tests 🧪 on some corner cases which are most likely accepted in E100, but a in the 'user error' territory. That by the way showed that the previous usage of Python's build-in `all()`-method actually didn't do what I intended it to do. Now using Pandas `all()`-method on the dataframe. So, something good came out of it 👍

**Review by berland — APPROVED** [review]
LGTM
